metadata/containers_test.go

```diff
@@ -420,6 +421,103 @@ func TestContainersCreateUpdateDelete(t *testing.T) {
 			},
 			createerr: errdefs.ErrInvalidArgument,
 		},
+		{
+			name: "UpdateExtensionsFull",
+			original: containers.Container{
+				Spec: encoded,
+				Runtime: containers.RuntimeInfo{
+					Name: "testruntime",
+				},
+				Extensions: &types.Any{
+					TypeUrl: "test.update.extensions",
+					Value:   []byte("hello"),
+				},
+			},
+			input: containers.Container{
+				Spec: encoded,
+				Runtime: containers.RuntimeInfo{
+					Name: "testruntime",
+				},
+				Extensions: &types.Any{
+					TypeUrl: "test.update.extensions",
+					Value:   []byte("world"),
+				},
+			},
+			expected: containers.Container{
+				Spec: encoded,
+				Runtime: containers.RuntimeInfo{
+					Name: "testruntime",
+				},
+				Extensions: &types.Any{
+					TypeUrl: "test.update.extensions",
+					Value:   []byte("world"),
+				},
+			},
+		},
+		{
+			name: "UpdateExtensionsNotInFieldpath",
+			original: containers.Container{
+				Spec: encoded,
+				Runtime: containers.RuntimeInfo{
+					Name: "testruntime",
+				},
+				Extensions: &types.Any{
+					TypeUrl: "test.update.extensions",
+					Value:   []byte("hello"),
+				},
+			},
+			input: containers.Container{
+				Spec: encoded,
+				Runtime: containers.RuntimeInfo{
+					Name: "testruntime",
+				},
+				Extensions: &types.Any{
+					TypeUrl: "test.update.extensions",
+					Value:   []byte("world"),
+				},
+			},
+			fieldpaths: []string{"labels"},
+			expected: containers.Container{
+				Spec: encoded,
+				Runtime: containers.RuntimeInfo{
+					Name: "testruntime",
+				},
+				Extensions: &types.Any{
+					TypeUrl: "test.update.extensions",
+					Value:   []byte("hello"),
+				},
+			},
+		},
+		{
+			name: "UpdateExtensionsFieldPath",
+			original: containers.Container{
+				Spec: encoded,
+				Runtime: containers.RuntimeInfo{
+					Name: "testruntime",
+				},
+				Extensions: &types.Any{
+					TypeUrl: "test.update.extensions",
+					Value:   []byte("hello"),
+				},
+			},
+			input: containers.Container{
```

Comment:
add something that can also be updated here to make sure it's not touched (e.g. `labels`) ?